### DISCUSSION TITLE
Manually set the selected item

### DISCUSSION THREAD

**Discussion by fiscio** [orchidjs/tom-select#58] [Q&A] [answered]
Hi there,

first of all, thanks to the devs for this great js component, that looks very promising.

I think I got it to work but still there is something, that sorta sounds rather basic, that I didn't get to work so far.

I have in the DOM a basic tom-select element, built the regular way, say:

`<select><optgroup><option></option></optgroup></select>`

1. The element is among other input fields that belong to a form within a Bootstrap _card_ component. 
2. Now I change the value of the tom-select field (by selecting one of the available options).
3. Let's imagine now that  I don't like to submit those changes, then I'd now need to "reselect" the previously selected option (meaning "resetting" the form to its previous state). For that, what I do is request the values I want via ajax and set them on the inputs upon ajax success.

But I haven't figured out yet how to do  it with the tom-select. The rest of the fields they work just fine.

I received the values via ajax in "room_id"  and "room_name" (which are the option "value" and option "text" respectively)

I tried SetValue() and setTextboxValue() but didn't pan out ...

** I'd figure it'd be something similiar to a regular Select input where you only need to set the option's value and that's it.

Thank you in advance. Hope someone understand and can help.

Cheers!

**Comment by oyejorge** [comment, marked as answer]
Yeah, based on your description I'd think setValue() would work. [Could you set up a jsfilddle to demonstrate your scenario](https://jsfiddle.net/a50w7x42/1/)?

**Reply by fiscio** [reply]
Hi,

thanks for your answer. I've tried again with setValue() and I finally got it to work but I had to "force" the value type by using parseInt() , otherwise didn't work.

From the values got via ajax ("location" var), I had to do:

`tomsels[tomsel_id].setValue(parseInt(location.room_id));`

and now it's working but I wonder why it does not work without using parseInt().  Any ideas?

Cheers!

**Reply by oyejorge** [reply]
Hmm, I'm not seeing a difference between setValue(1) and setValue("1"). Could you set up a jsfiddle?

**Reply by fiscio** [reply]
Hi, 

sorry for my late reply.

You're absolutely right, my bad! 
My mistake was a wrong evaluation on the server side...  I revisited my PHP code and it happens I was serving a string via ajax instead of an integer...

I'm sorry I wasted your time. 
Thanks again and keep up the good work!

** updated
What I meant is I don't really know exactly what I was doing because you're right and using a string or integer makes no difference at all. Sorry again.
